internal/metrics/bundledeployment_metrics.go

```diff
@@ -0,0 +1,74 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/rancher/fleet/internal/cmd/controller/summary"
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	bundleDeploymentSubsystem = "bundledeployment"
+	bundleDeploymentLabels    = []string{
+		"name",
+		"namespace",
+		"cluster_name",
+		"cluster_namespace",
+		"repo",
+		"commit",
+		"bundle",
+		"bundle_namespace",
+		"generation",
+		"state",
+	}
+	BundleDeploymentCollector = CollectorCollection{
+		subsystem: bundleDeploymentSubsystem,
+		metrics: map[string]prometheus.Collector{
+			"state": promauto.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Namespace: metricPrefix,
+					Subsystem: bundleDeploymentSubsystem,
+					Name:      "state",
+					Help: "Shows the state of this bundle deployment based on the state label. " +
+						"A value of 1 is true 0 is false.",
+				},
+				bundleDeploymentLabels,
+			),
+		},
+		collector: collectBundleDeploymentMetrics,
+	}
+)
+
+func collectBundleDeploymentMetrics(obj any, metrics map[string]prometheus.Collector) {
+	bundleDep, ok := obj.(*fleet.BundleDeployment)
+	if !ok {
+		panic("unexpected object type")
```

Comment:
I would generally agree that panic is bad, but...this would imply a code bug (connecting the wrong things) which means that it could be improved by making `CollectorCollection` generic with `[T runtime.Object]` ?